mtl/multitask_tutorial.py

```diff
@@ -29,81 +33,64 @@
 # Our data points are 2D points in a square centered on the origin.
 # Our tasks will be classifying whether these points are:
 #
-# 1. Inside a **unit circle** centered on the origin
-# 2. Inside a **unit square** centered on the origin
+# 1. Inside a **unit circle** centered on the origin (0 = False, 1 = True)
+# 2. Inside a **unit square** centered on the origin (0 = False, 1 = True)
 #
 # We'll visualize these decision boundaries in a few cells.
 #
-# _Note: In this toy example, we don't expect these specific tasks to help each other learn, but this is often a benefit of joint training in MTL settings._
+# _Note: In this toy example, we don't expect these specific tasks to help the model improve on the other task, but this is often a benefit of joint training in MTL settings._
+# %%
+import os
 
+# Make sure we're running from the spam/ directory
+if os.path.basename(os.getcwd()) == "snorkel-tutorials":
+    os.chdir("mtl")
 # %%
-import numpy as np
+from utils import make_circle_data, make_square_data
 
-N = 500  # Data points per dataset
+N = 1000  # Data points per dataset
 R = 1  # Unit distance
 
-# Dataset 0 ("circle")
-circle_data = np.random.uniform(0, 1, size=(N, 2)) * 2 - 1
-circle_labels = circle_data[:, 0] ** 2 + circle_data[:, 1] ** 2 < R
-
-# Dataset 1 ("square")
-square_data = np.random.uniform(0, 1, size=(N, 2)) * 2 - 1
-square_labels = (abs(square_data[:, 0]) < 0.5) * (abs(square_data[:, 1]) < 0.5)
+circle_data_splits, circle_label_splits = make_circle_data(N, R)
+square_data_splits, square_label_splits = make_square_data(N, R)
 
-# %% [markdown]
-# Note that, as is the case throughout the Snorkel repo, the **label -1 is reserved for abstaining/no label**; all actual labels have non-negative integer values: 0, 1, 2, .... This provides flexibility for supervision sources to label only portions of a dataset, for example.
+# %%
+print(f"Data splits: {list(circle_data_splits.keys())}")
+print(f"Training data shape: {circle_data_splits['train'].shape}")
+print(f"Label space: {set(circle_label_splits['train'])}")
 
 # %% [markdown]
 # And we can view the ground truth labels of our tasks visually to confirm our intuition on what the decision boundaries look like.
+# These two datasets occupy the same region in Euclidean space, but are disjoint sets of data points.
+# In the plots below, the purple points represent class 0 and the yellow points represent class 1.
 
 # %%
 import matplotlib.pyplot as plt
 
 fig, axs = plt.subplots(1, 2)
 
-axs[0].scatter(circle_data[:, 0], circle_data[:, 1], c=circle_labels)
+scatter = axs[0].scatter(
```

Comment:
This code is getting a bit too fancy and hard to read for a simple example. try to be more explicit and work with simple objects (e.g. ndarrays, not dictionaries of them)